pages/vomnibar.coffee

```diff
@@ -96,9 +96,25 @@ class VomnibarUI
       @input.value = @previousInputValue
       @previousInputValue = null
 
-    # Highlight the selected entry, and only the selected entry.
+    # Highlight the selected entry, and only the selected entry, and hide the entries too far away from the
+    # selected
+    [lowest, highest] = @calcRangeForVisibleChildren()
     for i in [0...@completionList.children.length]
-      @completionList.children[i].className = (if i == @selection then "vomnibarSelected" else "")
+      className = (if i == @selection then "vomnibarSelected" else if (lowest <= i < highest) then "" else "vomnibarHided")
+      @completionList.children[i].className = className
+
+  calcRangeForVisibleChildren: () ->
+    visibleRange = 10
+    if @completionList.children.length <= visibleRange
+      highest = @completionList.children.length
+      lowest = 0
+    else
+      highest = @selection+visibleRange//2+1
```

Comment:
Let's use `math.floor` rather than Coffeescript's obscure syntax. I didn't know what this was and had to look it up.